logger.js

```diff
@@ -46,13 +46,22 @@ function pinoLogger (opts, stream) {
     }, 'request completed')
   }
 
+  function onReqAborted (res) {
+    res.statusCode = 0
+    onResFinished.call(res, new Error('Aborted'))
+  }
+
   function loggingMiddleware (req, res, next) {
     req.id = genReqId(req)
     req.log = res.log = logger.child({req: req})
     res.startTime = Date.now()
 
     res.on('finish', onResFinished)
     res.on('error', onResFinished)
+    // it's possible that browser aborts connection, or http-server because of timeout
+    req.on('aborted', function () {
+      onReqAborted(res)
+    })
```

Comment:
we should avoid creating this closure if possible, not sure if there is a way. Maybe we can attach the response to the request (if it is not already there)